### PR TITLE
Refactor generated plans

### LINKED ISSUE #178: Refactor plan export
Move the rendering logic from the views so that exports can also be triggered via API, CLI, admin, pyshell.

### PR DESCRIPTION
**Pull request by hmpf**
Move the context-buildup and rendering out of the view so it can be
reused elsewhere.

Fixes #178

### CODE REVIEW

**Comment by hmpf** [comment]
The actual function could maybe be moved elsewhere: `easydmp.plan.utils`, say.

**Comment by hmpf** [comment]
We could also have fixed functions per format so that sending around the template name becomes unnecessary. Much is possible!

Anyway, the important point is disconnecting the rendering from the response, since the CLI need no response, and the API has its own response class.

File: src/easydmp/plan/views/__init__.py [hunk 1]
```diff
@@ -745,25 +758,21 @@ def get_queryset(self):
...
         template = '{timestamp} {actor} accessed generated version of {target}'
         log_event(request.user, 'access generated plan', target=self.object,
                   template=template)
-        return next
 
-    def get_editors(self):
-        return [access.user for access in
-                PlanAccess.objects.filter(plan=self.object).filter(may_edit=True).all()]
+    def get(self, request, *args, **kwargs):
+        self.generate_exported_plan()
```
Comment (vbhavdal) [inline src/easydmp/plan/views/__init__.py:772]: Well, do we need to store the export on self at all here? Isn't it cleaner to keep it function local?

Comment (hmpf) [reply src/easydmp/plan/views/__init__.py:772]: That method is strictly doing too much anyway but I wanted to keep `get` short and DRY. I had `log` in there for a while too.

I think I'll keep it as is, but the underlying wart is how get/post etc. works in CBVs at all, it mixes up too much stuff. What if you want to do something with the form-data before you clean, for instance.

**Pull request merged**